pkg/ephemeral/io/carrier.go

```diff
@@ -42,16 +46,47 @@ type Config struct {
 }
 
 // Connect establishes a TCP connection to a socket on a given host and port.
-func (c *Carrier) Connect(ctx context.Context, host, port string) error {
+func (c *Carrier) Connect(playerID int32, ctx context.Context, host string, port string) error {
 	conn, err := c.Dialer(ctx, host, port)
 	if err != nil {
 		return err
 	}
-	c.Conn = conn
+	_, err = conn.Write(c.buildHeader(playerID))
+	if err != nil {
+		return err
+	}
+	c.Conn, err = c.TlsConnector(conn, playerID)
+	if err != nil {
+		return err
+	}
+	if playerID == 0 {
+		err = c.readSpec()
+		if err != nil {
+			return err
+		}
+	}
 	c.connected = true
 	return nil
 }
 
+func (c Carrier) readSpec() error {
+	const size = 4
+	readBytes := make([]byte, size)
+	_, err := io.LimitReader(c.Conn, size).Read(readBytes)
+	if err != nil {
+		return err
+	}
+
+	sizeOfHeader := binary.LittleEndian.Uint32(readBytes)
+	readBytes = make([]byte, sizeOfHeader)
+	_, err = io.LimitReader(c.Conn, int64(sizeOfHeader)).Read(readBytes)
+	if err != nil {
+		return err
+	}
+	//ToDo, compare read PRIME with prime number from config?
```

Comment:
Is there a reason for this to be a ToDo

Reply:
It was more a question on whether or not we want to add this.
But that would probably require some more changes (so far ephemeral does not know the prime iirc).

Reply:
I think latest when tuples are streamed from castor, ephemeral needs to be aware of configuration parameters.
As it might not to be required yet, maybe we should at least create an issue for this one?

Reply:
Created #21 for this issue